lib/dynamoid/adapter.rb

```diff
@@ -19,7 +19,7 @@ def tables
       if !@tables_.value
         @tables_.swap{|value, args| benchmark('Cache Tables') {list_tables}}
       end
-      @tables_.value
+      @tables_.value || []
```

Comment:
Hey @adamthedeveloper what is the error that is returning without that `@tables_.value || []` ? could be this one: `Aws::DynamoDB::Errors::ResourceNotFoundException: Cannot do operations on a non-existent table`?
If that's the case has to be with the *caching* that Dynamoid is doing of the tables names.

What's your version of `aws-sdk-core` and `aws-sdk-resources` in order if we can maybe narrow the issue and I'm happy to built a test case for it if we found the scenario